src/dbwrapper.cpp

```diff
@@ -97,46 +97,59 @@ static void SetMaxOpenFiles(leveldb::Options *options) {
              options->max_open_files, default_open_files);
 }
 
-static leveldb::Options GetOptions(size_t nCacheSize)
+static void GetOptions(size_t nCacheSize, leveldb::Options* options)
 {
-    leveldb::Options options;
-    options.block_cache = leveldb::NewLRUCache(nCacheSize / 2);
-    options.write_buffer_size = nCacheSize / 4; // up to two write buffers may be held in memory simultaneously
-    options.filter_policy = leveldb::NewBloomFilterPolicy(10);
-    options.compression = leveldb::kNoCompression;
-    options.info_log = new CBitcoinLevelDBLogger();
+    options->block_cache = leveldb::NewLRUCache(nCacheSize / 2);
+    options->write_buffer_size = nCacheSize / 4; // up to two write buffers may be held in memory simultaneously
+    options->filter_policy = leveldb::NewBloomFilterPolicy(10);
+    options->compression = leveldb::kNoCompression;
+    options->info_log = new CBitcoinLevelDBLogger();
     if (leveldb::kMajorVersion > 1 || (leveldb::kMajorVersion == 1 && leveldb::kMinorVersion >= 16)) {
         // LevelDB versions before 1.16 consider short writes to be corruption. Only trigger error
         // on corruption in later versions.
-        options.paranoid_checks = true;
+        options->paranoid_checks = true;
     }
-    SetMaxOpenFiles(&options);
-    return options;
+    SetMaxOpenFiles(options);
+}
+
+// Custom deleter for unique_ptr<leveldb::Options>.
+static void DeleteOptions(leveldb::Options* options)
```

Comment:
In commit "dbwrapper: wrap options in unique_ptr and use DeleteOptions" (6770f066b69c6c271e7096cc647d5d42762f6b05)

Custom deleter seems to be forgetting to `delete options`, which would explain the fuzzing leak? https://github.com/bitcoin/bitcoin/pull/22663#issuecomment-894792761

Reply:
No longer using a custom deleter and instead using `~LevelDbOptions`